tests/test_requests.py

```diff
@@ -2540,6 +2540,28 @@ def test_json_encodes_as_bytes():
     assert isinstance(p.body, bytes)
 
 
+def test_data_ecodes_noniterables():
+    class Class:
+        pass
+
+    class ClassStringable:
+        def __str__(self):
+            return "🔥arbClassStringable🔥"
+
+    body = {
+        "string": "string",
+        "float": 0.01,
+        "int": 100,
+        "bool": True,
+        "None": None,
+        "plain_class": Class(),
```

Comment:
what does Requests do here?

Reply:
I don't think there any value having a str(nonstringableobj). we should do something about it.